src/test/java/com/google/javascript/clutz/partial/legacy_namespace_forward_declare.js

```diff
@@ -0,0 +1,13 @@
+goog.module('legacy.namespace.forward.declare');
+goog.module.declareLegacyNamespace();
```

Comment:
I keep forgetting, is the `declareLegacyNamespace` crucial to this test? Can you annotate the .d.ts output like 

//!! this block is needed for ...
declare namespace ಠ_ಠ.clutz.legacy.namespace.forward {

//!! this block is needed for
declare namespace ಠ_ಠ.clutz { ... }

Reply:
It is necessary, but in a tricky way.  I added a comment to the .d.ts.